### ISSUE TITLE
Proposal: allow a button attribute on form controls like <input>, <select>, <textarea> etc.

### ISSUE THREAD

**Issue by accessabilly** [w3c/html#1362]
Hi again dear W3C Team,

I currently have a problem that I wanted to solve via HTML, but there is no solution via standards, as far as I can see. 

I have a `<form>` that is wrapped around a `<table>`. Per table row, there are some form controls in different `<td>` elements like a `<select>` element and an `<input>` element and a `<button type="submit">` with it's own `formaction` attribute. There are the same controls per row with different `value`, but with the same `name`. My idea is to be able to submit just one row's data, although all row controls are in the same `<form>` and DOM at the same time. And because the `<form>` element can not be wrapped around a `<tr>`, I wrapped it around the `<table>`.

I know that I can associate `<input>` elements to a `<form>` element via the `form `attribute and so spread form controls all over the document, which is nice and handy btw!

But on the other hand, **it should be as well possible to assign a form control element to a specific `<button>`**, because a `<button>` can have a `formaction` attribute, which actually makes it a `<form>` itself, at least in my opinion.

I think of something like this:

```html
  <form method="get">

    <input type="text" button="MyButton1" name="foo" value="bar1">
    <button id="MyButton1" formaction="/route1">submit1</button>

    <input type="text" button="MyButton2" name="foo" value="bar2">
    <button id="MyButton2" formaction="/route2">submit2</button>

  </form>
```
So if I would click on submit1, the url would be like this: /route1?foo=bar1 
And clicking on submit 2, the url would be like this: /route2?foo=bar2 

In short: it would be nice, if I could have a lot of different **submit buttons** on one page and some **form controls that are associated by IDs to them.**

To make it a little more clear, I added a Codepen for that. Please check the Network in the Devtools for output, because it is trapped in an `<iframe>` ;-)
[Example](https://codepen.io/accessabilly/pen/oqVRoW)
You can see, that all form controls are submitted anytime you click confirm, and not the ones in a row, although they are are assigned to a different `formaction`, which was a wrong thougth of mine.

Greets and thank you for your work!
Martin

**Comment by chaals** [comment]
Hi @accessabilly, you can do something like what you want. You create forms, and associate the submit button and other controls with the `form` attribute, like so:

```
<form method="get" action="forms-in-tables-manual.html" id="form0"></form>
<form method="get" action="forms-in-tables-1-manual.html" id="form1">
<table>
 <tr>
  <td><label><input type="checkbox" name="zero" value="zero" checked form="form">zero</label>
  <td><input type="submit" form="form0">
 <tr>
  <td><label><input type="checkbox" name="one" value="one" checked>one</label>
  <td><input type="submit">
```

(the second row is inside its form so doesn't need the association through an attribute).

If this solves the issue I suggest closing it. Otherwise, I am misunderstanding so please explain it again...

**Comment by accessabilly** [comment]
Yes, I know that this is possible as well. It feels weird though, because I have a lot of empty `<form>` elements in the DOM. But I'm OK, if you are OK with that solution ;-)

Sorry for the inconvenience,
Martin
